include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -8821,4 +8821,21 @@ def err_bounds_type_annotation_lost_checking : Error<
   "type %diff{$ loses checking of declared type $|"
   "loses checking of declared type}0,1">;
 
+ def err_no_prototype_function_with_checked_return_type : Error<
+   "function without prototype cannot return a checked value">;
```

Comment:
Agreed.   Structs and unions make this somewhat difficult message to word: they aren't checked types, but they might contain checked types.  We might need two separate error message for this ultimately.